presto-main/src/main/java/io/prestosql/sql/gen/BytecodeUtils.java

```diff
@@ -345,6 +350,9 @@ else if (type == ConnectorSession.class) {
         if (functionMetadata.isNullable()) {
             block.append(unboxPrimitiveIfNecessary(scope, returnType));
         }
+        if (Object.class.equals(returnType)) {
+            block.comment("Cast to proper SQL return type").checkCast(sqlReturnType.getJavaType());
```

Comment:
was lack of this cast exploitable with any test today?

Reply:
After the change I made to `ArraySubscriptOperator`, I am not aware of any functions that require this cast, but I wouldn't be surprised if some existed.